tickets/src/components/content/NewEventContent.tsx

```diff
@@ -0,0 +1,178 @@
+
+interface UnlockWindow extends Window {
+  unlockProtocol?: any
+  unlockProtocolConfig?: any
+}
+
+declare var window: UnlockWindow | undefined
+
+enum PaywallStatus {
+  Locked = 'locked',
+  Unlocked = 'unlocked',
+  Unknown = 'unknown',
+}
+
+interface EventContentProps {
+  lockAddress: string
+  config: { env: string }
+  event: any
+}
+
+interface EventContentState {
+  paywallStatus: PaywallStatus
+}
+
+export class EventContent extends Component<
+  EventContentProps,
+  EventContentState
+> {
+  constructor(props: EventContentProps) {
+    super(props)
+
+    const { lockAddress } = props
+
+    this.state = {
+      paywallStatus: PaywallStatus.Unknown,
+    }
+
+    if (typeof window !== 'undefined') {
+      window.addEventListener('unlockProtocol', this.setPaywallStatus)
+
+      // TODO: Make this nicer
+      window.unlockProtocolConfig = {
+        persistentCheckout: false,
+        locks: {
+          [lockAddress]: {},
+        },
+        callToAction: {
+          default: 'Buy a ticket',
+        },
+      }
+    }
+  }
+
+  componentWillUnmount = () => {
+    if (window) {
+      window.removeEventListener('unlockProtocol', this.setPaywallStatus)
+    }
+  }
+
+  setPaywallStatus = (event: Event) => {
+    const d = (event as any).detail
+
+    // Making sure we only set state to expected values, since it's possible for
+    // someone to manually send a CustomEvent
+    if (d === PaywallStatus.Locked || d === PaywallStatus.Unlocked) {
+      this.setState({
+        paywallStatus: d,
+      })
+    }
+  }
+
+  paywallScriptUrl = (): string => {
+    const {
+      config: { env },
+    } = this.props
+    const extension = '/static/unlock.1.0.min.js'
+
+    // TODO: add test url?
+    const baseUrls: { [key: string]: string } = {
+      dev: 'http://localhost:3001',
+      staging: 'https://staging-paywall.unlock-protocol.com',
+      prod: 'https://paywall.unlock-protocol.com',
+    }
+
+    return baseUrls[env] + extension
+  }
+
+  render() {
+    const { paywallStatus } = this.state
+    const { event } = this.props
+
+    // TODO: make this better (get event faster? SSR?)
+    if (Object.keys(event).length === 0) {
+      return <Layout>Loading...</Layout>
+    }
+
+    return (
+      <Layout>
+        <Head>
+          <title>{pageTitle(event.name)}</title>
+          <script src={this.paywallScriptUrl()} />
+        </Head>
+        <EventInfo event={event} />
+        <Columns>
+          <Column>
+            <EventDescription body={event.description || ''} />
+            <EventLinks event={event} />
+          </Column>
+          <Column>
+            {paywallStatus === PaywallStatus.Locked && (
+              <PurchaseTicket
+                onClick={() => {
+                  if (typeof window !== 'undefined') {
+                    window.unlockProtocol &&
+                      window.unlockProtocol.loadCheckoutModal()
+                  }
+                }}
+                keyPrice="0.01"
```

Comment:
Is this place holder code until the next pr?

Reply:
It is, I plan on either
- getting the lock and key info out of the paywall directly
- or just hooking up to the lock information tickets currently passes through redux

Reply:
Yikes. This should have not been approved/merged as is.